src/xrdsvc/HttpWorkerCzarModule.cc

```diff
@@ -0,0 +1,213 @@
+        : HttpModule(context, foreman, req, resp) {}
+
+json HttpWorkerCzarModule::executeImpl(string const& subModuleName) {
+    string const func = string(__func__) + "[sub-module='" + subModuleName + "']";
+    enforceInstanceId(func, wconfig::WorkerConfig::instance()->replicationInstanceId());
+    enforceWorkerId(func);
+    if (subModuleName == "QUERYJOB") return _queryJob();
+    throw invalid_argument(context() + func + " unsupported sub-module");
+}
+
+json HttpWorkerCzarModule::_queryJob() {
+    debug(__func__);
+    checkApiVersion(__func__, 34);
+    // At this point, API version, correct worker, and auth have been checked.
+    json jsRet = _handleQueryJob(__func__);
+    return jsRet;
+}
+
+json HttpWorkerCzarModule::_handleQueryJob(string const& func) {
+    json jsRet;
+    vector<wbase::Task::Ptr> ujTasks;
+    try {
+        // See qdisp::UberJob::runUberJob() for json message construction.
+        auto const& jsReq = body().objJson;
+        string const targetWorkerId = body().required<string>("worker");
+
+        http::RequestBody rbCzar(body().required<json>("czar"));
+        auto czarName = rbCzar.required<string>("name");
+        auto czarId = rbCzar.required<qmeta::CzarId>("id");
+        auto czarPort = rbCzar.required<int>("management-port");
+        auto czarHostName = rbCzar.required<string>("management-host-name");
+        LOGS(_log, LOG_LVL_TRACE,
+             __func__ << " czar n=" << czarName << " id=" << czarId << " p=" << czarPort
+                      << " h=" << czarHostName);
+
+        http::RequestBody rbUberJob(body().required<json>("uberjob"));
+        auto ujQueryId = rbUberJob.required<QueryId>("queryid");
+        auto ujId = rbUberJob.required<UberJobId>("uberjobid");
+        auto ujCzarId = rbUberJob.required<int>("czarid");
+        auto ujJobs = rbUberJob.required<json>("jobs");
+        LOGS(_log, LOG_LVL_TRACE,
+             __func__ << " uj qid=" << ujQueryId << " ujid=" << ujId << " czid=" << ujCzarId);
+
+        auto ujData = wbase::UberJobData::create(ujId, czarName, czarId, czarHostName, czarPort, ujQueryId,
+                                                 targetWorkerId, foreman(), authKey());
+
+        // Find the entry for this queryId, creat a new one if needed.
+        wbase::UserQueryInfo::Ptr userQueryInfo = wbase::UserQueryInfo::uqMapInsert(ujQueryId);
+        userQueryInfo->addUberJob(ujData);
+
+        auto channelShared =
+                wbase::FileChannelShared::create(ujData, czarId, czarHostName, czarPort, targetWorkerId);
+        ujData->setFileChannelShared(channelShared);
+
+        // TODO:UJ These items should be stored higher in the message structure as they get
+        //   duplicated and should always be the same within an UberJob.
+        QueryId jdQueryId = 0;
+        proto::ScanInfo scanInfo;
+        bool scanInfoSet = false;
+        bool jdScanInteractive = false;
+        int jdMaxTableSize = 0;
```

Comment:
I would 100% agree with the comment posted above on:
```
        // TODO:UJ These items should be stored higher in the message structure as they get
        //   duplicated and should always be the same within an UberJob.
```
Why delaying its and not doing it right when it's clear what should be done?

Reply:
The answer to "Why delay?" is going to be there that there was a lot to be done (some of which is rather complicated) and getting a prototype working is more important than optimizing. Any time I move away from the original logic, there's a very real risk it would break something downstream, which then could break something else, and so on.